ligo/stablecoin/spec.ligo

```diff
@@ -0,0 +1,255 @@
+
+type token_metadata_params_ is record
+  token_ids : list (token_id)
+; callback  : contract (list (token_metadata_response))
+end
+
+type token_metadata_params is michelson_pair_right_comb(token_metadata_params_)
+
+type operator_tokens_ is
+| All_tokens
+| Some_tokens of set (token_id)
+
+type operator_tokens is michelson_or_right_comb(operator_tokens_)
+
+type operator_param_ is record
+  owner    : address
+; operator : address
+; tokens   : operator_tokens
+end
+
+type operator_param is michelson_pair_right_comb(operator_param_)
+
+type update_operator_param is
+| Add_operator    of operator_param
+| Remove_operator of operator_param
+
+type update_operator_params is list (update_operator_param)
+
+type is_operator_response_ is record
+  operator   : operator_param
+; is_operator : bool
+end
+
+type is_operator_response is michelson_pair_right_comb(is_operator_response_)
+
+type is_operator_params_ is record
+  operator : operator_param
+; callback : contract (is_operator_response)
+end
+
+type is_operator_params is michelson_pair_right_comb(is_operator_params_)
+
+(* ------------------------------------------------------------- *)
+
+type self_transfer_policy_ is
+| Self_transfer_permitted
+| Self_transfer_denied
+
+type self_transfer_policy is michelson_or_right_comb(self_transfer_policy_)
+
+type operator_transfer_policy_ is
+| Operator_transfer_permitted
+| Operator_transfer_denied
+
+type operator_transfer_policy is michelson_or_right_comb(operator_transfer_policy_)
+
+type owner_transfer_policy_ is
+| Owner_no_op
+| Optional_owner_hook
+| Required_owner_hook
+
+type owner_transfer_policy is michelson_or_right_comb(owner_transfer_policy_)
+
+type custom_permission_policy_ is record
+  tag        : string
+; config_api : option (address)
+end
+
+type custom_permission_policy is michelson_pair_right_comb(custom_permission_policy_)
+
+type permissions_descriptor_ is record
+  self     : self_transfer_policy
+; operator : operator_transfer_policy
+; receiver : owner_transfer_policy
+; sender   : owner_transfer_policy
+; custom   : option (custom_permission_policy)
+end
+
+type permissions_descriptor is michelson_pair_right_comb(permissions_descriptor_)
+
+type permissions_descriptor_params is
+  contract (permissions_descriptor)
+
+type mint_params is record
+  to_    : address
+; amount : nat
+end
+
+type burn_params is record
+  from_  : address
+; amount : nat
+end
+
+type parameter is
+  Transfer               of transfer_params
+| Balance_of             of balance_of_params
+| Total_supply           of total_supply_params
+| Token_metadata         of token_metadata_params
+| Permissions_descriptor of permissions_descriptor_params
+| Update_operators       of update_operator_params
+| Is_operator            of is_operator_params
+
+(* ------------------------------------------------------------- *)
+
+(*
+ * Hooks
+ *)
+
+// TODO: currently it's the same as `transfer` but with optional
+// participants.
```

Comment:
That's how it is specified in FA2. Do you think we can do anything about it?

Reply:
No, just wanted to mention that we use it only in `validate_owner_hook`

Reply:
Okay, then I think it shouldn't be a TODO. Maybe just an ordinary comment.